skyportal/handlers/api/comment.py

```diff
@@ -159,3 +162,35 @@ def delete(self, comment_id):
         self.push_all(action='skyportal/REFRESH_SOURCE',
                       payload={'source_id': source_id})
         return self.success()
+
+
+class CommentAttachmentHandler(BaseHandler):
+    @auth_or_token
+    def get(self, comment_id):
+        """
+        ---
+        description: Download comment attachment
+        parameters:
+          - in: path
+            name: comment_id
+            required: true
+            schema:
+              type: integer
+        responses:
+          200:
+            content:
+              application:
+                schema:
+                  type: string
+                  format: base64
+                  description: base64-encoded contents of attachment
+        """
+        comment = Comment.query.get(comment_id)
+        if comment is None:
+            return self.error('Invalid comment ID.')
+        # Ensure user/token has access to parent source
+        s = Source.get_if_owned_by(comment.source.id, self.current_user)
```

Comment:
This source is fetched, but not used?

Reply:
Yes, did you see the comment above? It's for permissions checking

Reply:
Hrm, does "get_if_owned_by" throw an error if it fails, or how do you report that back to the user?

Reply:
Yeah, it returns with an informative error message